feat: enable bossbar by default & support for custom title texts

Phoenix616 opened the pull request:
### **User description**
This fixes that the bossbar would not be rendered properly when added via the `/bossbar` command/plugin API and wasn't using the component rendering system but manual mapping. Also updates to the boss bar would not get rendered before.


___

### **PR Type**
Bug fix, Enhancement


___

### **Description**
- Fixed boss bar titles not displaying correctly.

- Updated boss bar title handling to support various formats.

- Refactored boss bar management with dedicated add/remove functions.

- Improved boss bar update logic for better rendering.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>BossBarOverlay.tsx</strong><dd><code>Refactor boss bar title handling and styles</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/react/BossBarOverlay.tsx

<li>Changed boss bar title handling to support multiple formats.<br> <li> Removed manual translation mapping for titles.<br> <li> Updated state initialization for boss bar title.<br> <li> Adjusted logic for setting boss bar styles.


</details>


  </td>
  <td><a href="https://github.com/zardoy/minecraft-web-client/pull/259/files#diff-2587f5ae9097b8c4f5d91303e8a4525d90e2a7a9039b211c92813c5c90e2dc88">+4/-8</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>BossBarOverlayProvider.tsx</strong><dd><code>Refactor boss bar management with utility functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/react/BossBarOverlayProvider.tsx

<li>Added <code>addBossBar</code> and <code>removeBossBar</code> utility functions.<br> <li> Simplified boss bar event handling with new functions.<br> <li> Improved boss bar update logic with a timeout mechanism.


</details>


  </td>
  <td><a href="https://github.com/zardoy/minecraft-web-client/pull/259/files#diff-35c17982c86cfee395c5650d36a988e30649b57d80357c5133498e869f6bdb82">+15/-5</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

Phoenix616 committed:
fix: bossbar does not display custom titles

zardoy commented on src/react/BossBarOverlayProvider.tsx:
```diff
@@ -5,18 +5,28 @@ import './BossBarOverlay.css'
     bot.on('bossBarUpdated', (bossBar) => {
-      setBossBars(prevBossBars => new Map(prevBossBars.set(bossBar.entityUUID, bossBar as BossBarType)))
+      removeBossBar(bossBar as BossBarType)
+      setTimeout(() => addBossBar(bossBar as BossBarType), 1)
```
why do we need timeout? would not create it flickering, so weird

Phoenix616 replied:
It will otherwise not update the displayed text/values/styles, most likely because the element already exists? It was the only solution I found but I don't know how react works so there might be a better way to force an update of the changed element.

zardoy reviewed (changes requested):
cant get it working, @Phoenix616 please specify versions on which you test your fixes

zardoy commented:
ok, my apologies, it is not visible only after rejoining.

though IMO further polishing is possible (wdyt?):
current:
![image](https://github.com/user-attachments/assets/2c18513a-7044-4dc8-aba2-5abb93dde619)
vanilla: ![image](https://github.com/user-attachments/assets/1428b860-3647-40ff-833c-d3d4c6ce44fa)
mobile current:
![image](https://github.com/user-attachments/assets/090559cc-8b8a-4c31-aa54-cccd2049259c)


IMO need to:
bossbar-title class: set font size to 9px (make a bit bigger), add margin-bottom to at least 1px (can leave as 1px for now)
when touch active: .bossBars top = 18px , when not 0px

Phoenix616 commented:
Style changes make sense, yeah.

I'm unsure what you mean by "lets enable them by default" though. I didn't adjust any code regarding the behavior or when and how it displays, it instantly shows them as soon as the mineflayer bot event is called which should be whenever a packet is received.

zardoy commented:
> Style changes make sense, yeah.
> 
> I'm unsure what you mean by "lets enable them by default" though. I didn't adjust any code regarding the behavior or when and how it displays, it instantly shows them as soon as the mineflayer bot event is called which should be whenever a packet is received.

no worries I can these changes myself, I just have hundreds of tasks that I try to do at the same and had to write these here so I don't forget, and also wanted to know your opinion on these changes

Phoenix616 committed:
fix: style of the boss bars not matching and mobile issues

- Adjusts the font size and gabs
- Fixes that it would overlap the mobile buttons
- Fixes that touching where the boss bar is would not move the camera
- Better match the behaviour of really wide bossbar titles

Phoenix616 commented:
Did some adjustments to the bossbar styles to make them behave closer to Java.

zardoy commented:
@Phoenix616 again, i still do not understand the intention to for setTimeout, will ltry to figure out as well will try to fix the re-login issue

zardoy committed:
enable bossbars

zardoy commented:
ok dont have time, feel free to help, need a plugin in src/mineflayer/plugins/bossbars.ts so bossbars are preserved into mineflayer bot state

zardoy committed:
fix mobile class
